src/librustc_mir/transform/const_prop.rs

```diff
@@ -905,6 +933,16 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
         }
     }
 
+    fn visit_operand(&mut self, operand: &mut Operand<'tcx>, location: Location) {
+        // Only const prop copies and moves on `mir_opt_level=3` as doing so
+        // currently increases compile time.
+        if self.tcx.sess.opts.debugging_opts.mir_opt_level < 3 {
+            self.super_operand(operand, location)
+        } else {
+            self.propagate_operand(operand, location)
```

Comment:
Why does this not call `super_operand`? Almost all visitors do that; there should be a comment explaining why it does not happen here.

Reply:
aren't we doing this inside of `propagate_operand`? I am not too familiar with this code though?

Should I rewrite it to
```rust
        if self.tcx.sess.opts.debugging_opts.mir_opt_level >= 3 {
            self.propagate_operand(operand, location)
        }
        self.super_operand(operand, location)

Reply:
> aren't we doing this inside of propagate_operand?

You added that function in this PR, didn't you? I cannot see it doing anything like that, but maybe I am missing the right line.

> Should I rewrite it to

I think that would be better, yes.

Reply:
done